src/cross_section/fillet2d.cpp

```diff
@@ -0,0 +1,661 @@
+
+    if (!p2IsConvex) {
+      // Handle concave endpoint
+      vec2 circleOffsetP2 = p2 + e1 * 2.0 * radius + normal * 2.0 * radius,
+           circleOffsetP2N = p2 + e1 * 2.0 * radius - normal * 2.0 * radius;
+
+      box.Union(toVec3(circleOffsetP2));
+      box.Union(toVec3(circleOffsetP2N));
+    }
+
+    auto r = collider.outerCollider.Collisions(
+        manifold::Vec<manifold::Box>({box}).cview());
+    // r.Dump();
+    r.Sort();
+
+    std::cout << "Now " << p1i << "->" << (e1i + 1) % outerLoop.size()
+              << std::endl;
+
+    // In Out Classify
+    for (size_t j = 0; j != r.size(); j++) {
+      auto ele = collider.outerPair[r.Get(j, true)];
+
+      size_t e2i = ele.p1Ref;
+
+      // Skip self and last one
+      if ((e1i == e2i) ||
+          e2i == (e1i + outerLoop.size() - 1) % outerLoop.size())
+        continue;
+
+      // CCW, p3 p4 -> bbox hit edge start end
+      size_t p3i = e2i, p4i = ele.p2Ref;
+      vec2 p3 = outerLoop[p3i], p4 = outerLoop[p4i];
+      vec2 t;
+
+      bool segmentIntersected = intersectSegmentSegment(
+               normalOffsetP1, normalOffsetP2, p3, p4, t),
+           circleIntersected =
+               p2IsConvex ? intersectCircleSegment(p3, p4, p2, radius) : false;
+      // FIXME: intersectCircleSegment start rad and end rad
+
+      std::cout << "Testing " << p3i << "->" << p4i << "\t";
+
+      std::cout << (segmentIntersected ? "Segment Intersected" : "Segment - ")
+                << " "
+                << (circleIntersected ? "Circle Intersected" : "Circle -");
+
+      double e1T = 0, e2T = 0;
+      vec2 circleCenter(0, 0);
+
+      if (!segmentIntersected && !circleIntersected) {
+        std::cout << std::endl;
+        continue;
+      } else if (circleIntersected) {
+        // Concave p2 endpoint intersected, degenerated natively
+
+        if (!calculatePointSegmentCircleCenter(p3, p4, p2, radius, e1T,
+                                               circleCenter)) {
+          continue;
+        }
+      } else if (segmentIntersected) {
+        if (!calculateSegmentSegmentCircleCenter(p1, p2, p3, p4, radius, e1T,
+                                                 e2T, circleCenter)) {
+          continue;
+        } else {
+          // Check if is endpoint, and add duplicate mark
+          if (e1T == 0) {
+            // p1
+            if (markEV[e2i * outerLoop.size() + p1i]) continue;
+            markEV[e2i * outerLoop.size() + p1i] = 1;
+          } else if (e1T == 1) {
+            // p2
+            if (markEV[e2i * outerLoop.size() + p2i]) continue;
+            markEV[e2i * outerLoop.size() + p2i] = 1;
+          }
+
+          if (e2T == 0) {
+            if (markEV[e1i * outerLoop.size() + p3i]) continue;
+            markEV[e1i * outerLoop.size() + p3i] = 1;
+          } else if (e2T == 1) {
+            if (markEV[e1i * outerLoop.size() + p4i]) continue;
+            markEV[e1i * outerLoop.size() + p4i] = 1;
+          }
```

Comment:
`calculateSegmentSegmentCircleCenter` will return result with multi situation, and need be process according to it. Current way is pretty unclean here, is there a better way to do it?